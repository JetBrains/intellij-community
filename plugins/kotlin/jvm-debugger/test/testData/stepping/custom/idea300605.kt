// ATTACH_LIBRARY: maven(org.jetbrains.kotlinx:kotlinx-coroutines-core:1.4.2)

package idea300605

fun test1() {
    // STEP_INTO: 1
    // RESUME: 1
    //Breakpoint!
    println()
}

class Foo(val d: Double) {
    fun bar() = println(d)
}

fun test2() {
    val foo = Foo(2.7)
    // STEP_INTO: 2
    // RESUME: 1
    //Breakpoint!
    foo.bar()
    println()
}

fun test3() {
    val lv = "local val is important here"
    fun localFun() {
        println("localFun() + $lv")
    }

    localFun()
    // STEP_INTO: 1
    // RESUME: 1
    //Breakpoint!
    val newName = ::localFun
    newName()
}

fun test4() {
    // STEP_INTO: 1
    // RESUME: 1
    //Breakpoint!
    listOf<Int>().extGenericFun()
}

fun <T> List<T>.extGenericFun(): List<T> = this


fun main() {
    test1()
    test2()
    test3()
    test4()
}
