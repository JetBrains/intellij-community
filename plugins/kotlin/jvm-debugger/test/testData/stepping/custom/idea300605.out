LineBreakpoint created at idea300605.kt:9
LineBreakpoint created at idea300605.kt:21
LineBreakpoint created at idea300605.kt:35
LineBreakpoint created at idea300605.kt:43
Run Java
Connected to the target VM
idea300605.kt:9
idea300605.kt:10
idea300605.kt:21
idea300605.kt:13
idea300605.kt:22
idea300605.kt:35
idea300605.kt:36
idea300605.kt:43
idea300605.kt:46
Disconnected from the target VM

Process finished with exit code 0
